### PR TITLE
[WIP]feat: Prediction

### PR DESCRIPTION
**Pull request by pancake-cat**
[ ] Before opening a pull request, please read the [contributing guidelines](https://github.com/pancakeswap/pancake-frontend/blob/master/CONTRIBUTING.md) first
[ ] If your PR is work in progress, open it as `draft`
[ ] Before requesting a review, all the checks need to pass
[ ] Explain what your PR does

### CODE REVIEW

File: src/views/Prediction/components/CountDown.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import { Text, TextProps } from '@pancakeswap-libs/uikit'
+import { useTimer } from 'hooks/useTimer'
+
+interface CountDownProps extends TextProps {
+  seconds: number
+  timeOver?: () => void
+  style?: Record<string, string>
```
Comment (hachiojidev) [inline src/views/Prediction/components/CountDown.tsx:8]: This should be `React.CSSProperties`

File: src/views/Prediction/components/CountDown.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  const h = Math.floor(l / 3600)
+  l %= 3600
+  const m = Math.floor(l / 60)
+  l %= 60
+  return {
+    h: `${h}`.padStart(2, '0'),
+    m: `${m}`.padStart(2, '0'),
+    s: `${l}`.padStart(2, '0'),
+  }
+}
```
Comment (hachiojidev) [inline src/views/Prediction/components/CountDown.tsx:22]: Please use `getTimePeriods`

File: src/views/Prediction/components/CountDown.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+const CountDown: React.FC<CountDownProps> = ({ seconds, timeOver, ...restProps }) => {
+  const _seconds = seconds > 0 ? seconds : 0
```
Comment (hachiojidev) [inline src/views/Prediction/components/CountDown.tsx:25]: Please remove the underscore.

File: src/views/Prediction/components/PredictionItem.tsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+  flex-shrink: 0;
+`
+const CardHeader = styled.div<{ status: STATUS }>`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  height: 28px;
+  padding: 0 16px;
+  background-color: ${({ theme, status }) => theme.colors[ConfigMap[status].bg]};
+`
```
Comment (hachiojidev) [inline src/views/Prediction/components/PredictionItem.tsx:48]: Please use the UI Kit's `<Flex />` component

File: tsconfig.json [hunk 5]
```diff
@@ -18,7 +18,7 @@
     "resolveJsonModule": true,
     "isolatedModules": true,
     "noEmit": true,
-    "jsx": "react-jsx",
+    "jsx": "react",
```
Comment (hachiojidev) [inline tsconfig.json:21]: This is incorrect. Please update your IDE's typescript version and revert this.

File: src/views/Prediction/components/TradingView.tsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+import styled from 'styled-components'
+import TradingViewWidget, { Themes } from 'react-tradingview-widget'
+
+const ChartCard = styled.div`
+  flex: 1;
+  margin: 24px 0 4px 0;
+`
```
Comment (hachiojidev) [inline src/views/Prediction/components/TradingView.tsx:8]: Please use the Flex component

File: src/views/Prediction/components/PredictionList.tsx [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+      { status: 'COMING' },
+      { status: 'COMING' },
+    ])
+  }, [setPredictions])
+
+  return (
+    <Container>
+      <ScrollContainer>
+        {predictions.map((item) => (
+          <PredictionItem {...item} />
```
Comment (hachiojidev) [inline src/views/Prediction/components/PredictionList.tsx:39]: This needs a key

**Review by hachiojidev — CHANGES_REQUESTED** [review]
A few things I noticed.

File: src/hooks/useMarketTrade.ts [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import { useState, useEffect } from 'react'
+import { marketStore } from 'utils/websocket-stream'
+
+function useMarketTarde() {
+  const [market, setMatket] = useState({ price: '', change: '' })
+
+  useEffect(() => {
+    return marketStore.subscribe(setMatket)
+  }, [])
```
[the remark this answers is not in the document]
Comment (pancake-cat) [reply src/hooks/useMarketTrade.ts]: yes

File: src/hooks/useTimer.ts [hunk 9]
```diff
@@ -16,4 +16,28 @@ export const useCurrentTime = () => {
   return currentMillis
 }
 
+export const useTimer = (initSecond) => {
+  const [remine, setRemine] = useState(initSecond)
```
[the remark this answers is not in the document]
Comment (pancake-cat) [reply src/hooks/useTimer.ts]: a typo issue, fixed, thank you

**Pull request closed without merging**